### ISSUE TITLE
Sourcery unnesting function

### ISSUE THREAD

**Issue by harvey251** [sourcery-ai/sourcery#186]
### Issue description or question
When `Hoist repeated code outside conditional statement, Merge else clause's nested if statement into elif` it appears to un nest my function

```
<html><b>state_maker_index</b>   Hoist repeated code outside conditional statement, Merge else clause's nested if statement into elif</html>
<html><body><pre>             version_config[&quot;version&quot;] = 1
             version_config[&quot;revision&quot;] = 0
             version_config[&quot;status&quot;] = &quot;IN_PROCESS&quot;
<span style='color:#AA4926'>-            version_config = DataVersionTableModel(**version_config)</span>
<span style='color:#AA4926'>-        else:</span>
<span style='color:#AA4926'>-            if key not in version_config:</span>
<span style='color:#AA4926'>-                version_config[&quot;key&quot;] = key</span>
<span style='color:#AA4926'>-            version_config = DataVersionTableModel(**version_config)</span>
<span style='color:green'>+        elif key not in version_config:</span>
<span style='color:green'>+            version_config[&quot;key&quot;] = key</span>
<span style='color:green'>+        version_config = DataVersionTableModel(**version_config)</span></pre></body></html>
```

```
@pytest.fixture
def state_maker_index(mocker):
    def _state_maker_index(
        key=None, metadata_config=None, version_config=None
    ):
        if version_config is None:
            version_config = {}
            version_config["key"] = metadata_config["key"]
            version_config["version"] = 1
            version_config["revision"] = 0
            version_config["status"] = "IN_PROCESS"
            version_config = DataVersionTableModel(**version_config)
        else:
            if key not in version_config:
                version_config["key"] = key
            version_config = DataVersionTableModel(**version_config)

    return _state_maker_index
```

AFTER 

```
def _state_maker_index(
        key=None, metadata_config=None, version_config=None
    ):
        if version_config is None:
            version_config = {}
            version_config["key"] = metadata_config["key"]
            version_config["version"] = 1
            version_config["revision"] = 0
            version_config["status"] = "IN_PROCESS"
        elif key not in version_config:
            version_config["key"] = key
        version_config = DataVersionTableModel(**version_config)
 ```

### Sourcery Version

Often available in Sourcery settings
0.10.0

### Code editor or IDE name and version

PyCharm 2021.3.1 (Professional Edition)

### OS name and version

Linux Ubuntu 20.04.3 LTS

**Comment by Hellebore** [comment]
Thanks for raising - yes clear bug - we will address!

**Comment by brendanator** [comment]
This looks like the same issues as https://github.com/sourcery-ai/sourcery/issues/193

It will be fixed in the next release. Thanks for reporting!

**Comment by Hellebore** [comment]
Fixed with release of 0.11.1
